docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+                remember_me_parameter: autologin
+
+            logout:
+                path: contao_frontend_logout
+                target: contao_root
+                handlers:
+                    - contao.security.logout_handler
+                success_handler: contao.security.logout_success_handler
+
+    access_control:
+        - { path: ^/contao/login$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao/logout$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao(/|$), roles: ROLE_USER }
+        - { path: ^/, roles: [IS_AUTHENTICATED_ANONYMOUSLY] }
+```
+{{% /expand %}}
+
+If you want to learn more about Symfony's Security Component use the provided links to read up on. This documentation will only cover
+implmentation details that are unique to Contao.
+
+Since within Contao you can put a login form on basically any page, Contao does not utilise Symfony's built-in 
+[`form_login` Authentication Provider][SymfonyFormLogin]. Instead, Contao implements its own user providers, 
+[user checkers][SymfonyUserChecker] and [request matcher service][SymfonyRequestMatcherService], the latter of which checks the 
+[request scope][RequestScope]. For example in the front end, all URLs to Contao pages will have the `_scope` request attribute set to 
+`frontend` and the `contao_frontend` firewall will thus be applicable to all these URLs.
+
+
+## Voters
+
+Starting with Contao **4.7** Contao implements so called [Voters][SymfonyVoters] in order to easily check whether an authenticated user is
+authorized to access specific resources. These voters are automatically used by Symfony via the [Security Helper][SecurityHelperService].
+
+{{< version "4.7" >}}
+
+The security helper can be used to check whether the currently authenticated user has access in the back end to specific forms, table fields
+(as defined via the DCA), folders and modules for example:
+
+```php
+// User has access to form ID 5
+$security->isGranted('contao_user.forms', 5);
+
+// User is allowed to access field "published" of table "tl_page"
+$security->isGranted('contao_user.alexf', 'tl_page::published');
+
+// Check access to folder
+$security->isGranted('contao_user.filemounts', '/files/foo/bar');
+```
+
+{{< version "4.10" >}}
+
+You can also use the security helper to check wether the user can edit a page or is allowed to edit fields in a specific DCA for example:
+
+```php
+// whether the user can access any field in tl_page
+$security->isGranted('contao_user.can_edit_fields', 'tl_page');
+
+// whether user can edit the given page (array of row or page model)
+$security->isGranted('contao_user.can_edit_page', [/* row of page data */]);
+$security->isGranted('contao_user.can_edit_page', $pageModel);
+```
+
+{{< version "4.12" >}}
+
+The security helper can also be used to check whether a front end or back end user belongs to any of the specified user groups:
+
+```php
+$security->isGranted('contao_group', $groupId);
```

Comment:
The string constant in `ContaoCorePermissions` is `'contao_member.groups'`, not `'contao_group'`.